decidim-conferences/spec/forms/registration_type_form_spec.rb

```diff
@@ -91,7 +91,9 @@ module Admin
         context "when price is not number" do
           let(:price) { "blabla" }
 
-          it { is_expected.to be_invalid }
+          # The value is cast to the correct type so "blabla" becomes 0.0 which
```

Comment:
Yes, exactly this is the reason. Same thing for `ActiveModel::Type::Integer.new.cast("blabla")`.

I don't think it's that bad actually as most of the times those inputs should be `<input type="number">` which shouldn't allow adding other inputs than numbers. Firefox is an exception to that rule but I don't think it's that big of an issue in actual use cases that `blabla` (or any string) would be converted to zeros as that input should be disabled by the UI in the ideal situation (such as Chrome).

Empty string input will still cast to `nil` which allows adding empty values to these fields.

Reply:
Ok, fair enough, for me the only problem is how the spec is read:

```ruby
let(:price) { "blabla" }
it { is_expected.to be_valid }
```
But with your comment, it is clear that this is a weird behavior indeed.

Also, I've checked it and for end users there isn't any behavior change, when adding a string in this field in Firefox it saves it with as an empty string on both cases (`Virtus` and `Attributes` API), so that's fine for me.